### ISSUE TITLE
GetItemLinqQueryable and ResourceToken does not work with cross partition query

### ISSUE THREAD

**Issue by fuocor** [Azure/azure-cosmos-dotnet-v3#1028]
**Describe the bug**
If you create Partition Permission  and create a cross-partition  Query on the container, it will throw an ```UnauthorizedException``` from ```IAuthorizationTokenProvider.GetUserAuthorizationToken```

**To Reproduce**
```
var result = await tokenContainer.GetItemLinqQueryable<IUserDefinition>()
                        .Where(i => i.Id == userDefinition2.Id)
                        .ToStreamIterator();
```

**Expected behavior**
return empty result. The caller should not need Read permission on the container to execute the statement.

**Actual behavior**
```UnauthorizedException```

**Environment summary**
SDK Version: Source
OS Version (e.g. Windows 10

**Comment by j82w** [comment]
Does it work if you set the QueryRequestOptions.PartitionKey to the Partition you gave it permission to?

**Comment by j82w** [comment]
@fuocor fixed with PR #1143
